uaclient/timer/update_messaging.py

```diff
@@ -125,17 +128,16 @@ def update_motd_messages(cfg: UAConfig) -> bool:
         service = "n/a"
         pkg_num = 0
 
+        enabled_services = _enabled_services(cfg).enabled_services
         if system.is_current_series_active_esm():
-            esm_infra_status, _ = ESMInfraEntitlement(cfg).application_status()
-            if esm_infra_status == ApplicationStatus.ENABLED:
-                service = "esm-infra"
+            service = "esm-infra"
+            if any(ent.name == "esm-infra" for ent in enabled_services):
```

Comment:
I think setting `service` needs to happen inside the if statement like before.

Separately, an alternative is-enabled implementation that looks very slightly better to me but not a strong opinion
```python
enabled_services_names = {s.name for s in _enabled_services(cfg).enabled_services}
if "esm-apps" in enabled_services_names:
    # continue
```

Reply:
These comments apply right below to esm-apps as well